api/src/wfl/module/covid.clj

```diff
@@ -96,27 +97,40 @@
 ;  ""
 ;  [])
 
+
 (defn verify-source!
+  "Verify that the `source` is named Terra DataRepo.
+   Verify that WFL has the necessary permissions to
+   read from the `dataset` and that it exists."
   [{:keys [name dataset] :as source}]
   (when-not (= (:name source) "Terra DataRepo")
     (throw (ex-info "Unknown Source" {:source source})))
-  ;; TODO: verify workspace exists
   (try
-    (datarepo/dataset (:dataset source))
-    (catch Throwable _
-      (throw (ex-info "Cannot access Dataset" {:dataset dataset})))
-    )
-  )
+    (#((datarepo/dataset (:dataset source)) nil))
```

Comment:
Can someone explain to me the change to anonymous functions here?

Reply:
See my comments in the other place.